assets/js/blocks/product-category/block.js

```diff
@@ -48,6 +53,36 @@ class ProductByCategoryBlock extends Component {
 		}
 	}
 
+	startEditing() {
+		this.setState( {
+			isEditing: true,
+			changedAttributes: {},
+		} );
+	}
+
+	stopEditing() {
+		this.setState( {
+			isEditing: false,
+			changedAttributes: null,
+		} );
+	}
+
+	setAttributes( attributes ) {
```

Comment:
I used `setAttributes` but I think there is a global/inherited setAttributes too.. I'm not sure where that is defined. Should I use another name? (this method updates the state, rather than the block attributes directly)

Reply:
Yes, `setAttributes` is one of the props this component receives, you can see props definition at the end of the file, under [`propTypes`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/583/files#diff-1f0006ba8a586d7d5f9939781d32257fR309).

I think it's usually better not to duplicate prop names as method names to avoid confusion (now we have `this.props.setAttributes` and `this.setAttributes`, which do different things but have the same name). For example, I would rename this method to `setChangedAttributes` or `updateChangedAttributes` or something like that.

Reply:
I agree that this name `this.setAttributes` should be changed, either of the suggestions above work 👍